### DISCUSSION TITLE
Can't make console warning about feature flag globals disapear

### DISCUSSION THREAD

**Discussion by wladston** [intlify/vue-i18n#914] [Q&A] [answered]
I have this warning in my console in dev mode:

`You are running the esm-bundler build of vue-i18n. It is recommended to configure your bundler to explicitly replace feature flag globals with boolean literals to get proper tree-shaking in the final bundle.`

I can't get rid of it…

I'm currently in the process of migrating my app from Webpack to Vite. In webpack, I had this in my config:

```
        plugins: [

            new webpack.DefinePlugin({
                __VUE_OPTIONS_API__: true,
                __VUE_PROD_DEVTOOLS__: false,
                __VUE_I18N_FULL_INSTALL__: true,
                __VUE_I18N_LEGACY_API__: true,
                __INTLIFY_PROD_DEVTOOLS__: false
            }),
```

That made the warning disapear. Now when using Vite, the warning is back, despite [the documentation saying these variables are automatically set when using Vite](https://vue-i18n.intlify.dev/guide/advanced/optimization.html#reduce-bundle-size-with-feature-build-flags). So I tried to add the variables on Vite, but it didn't make the warning go away. My `vite.config.js` looks like this:

```
export default {
    define: {
        __VUE_OPTIONS_API__: true,
        __VUE_PROD_DEVTOOLS__: false,
        __VUE_I18N_FULL_INSTALL__: true,
        __VUE_I18N_LEGACY_API__: true,
        __INTLIFY_PROD_DEVTOOLS__: false
    },
    …
```

I tried replacing the booleans for strings (`'true'` instead of `true`) and it also didn't work.

How can I get rid of the warning in the console?

I'm on `vue-i18n@9.1.9` and `@intlify/vite-plugin-vue-i18n@3.2.1`. I spend some time trying to build a minimal example of the problem, but I coudn't find an example boilerplate to use…

**Comment by kazupon** [comment, marked as answer]
vue-i18n v9.1.9 has a bug that cannot remove warning.
You can avoid it with using the vue-i18n v9.2-beta latest version

**Reply by wladston** [reply]
Thanks! It worked :) I just had to add an `overrides` key to my package.json and it worked!

```
  "overrides": {
    "@intlify/vite-plugin-vue-i18n" : {
      "vue-i18n": "9.2.0-beta.30"
    }
  },
```
